ext/ruby.c

```diff
@@ -4,3 +4,6 @@
 VALUE RS_Qtrue = Qtrue;
 VALUE RS_Qfalse = Qfalse;
 VALUE RS_Qnil = Qnil;
+
+long RS_RSTRING_LEN(VALUE str) { return RSTRING_LEN(str); }
+char* RS_RSTRING_PTR(VALUE str) { return RSTRING_PTR(str); }
```

Comment:
`const char*`

Reply:
Isn't it technically a `char *`?

Reply:
It should match the Rust binding; if it's a `char *` here it should be `*mut c_char`. It is basically one of the few places the C compiler _sort of_ checks mutability for you. However, I don't know if using this as a `*mut` is really supported. It seems pretty sketchy to me, so I think it's better to start off with `const` and relax it if people complain (at the end of the day, the `const char*` and `char*` doesn't really make a difference when calling from Rust, it's really just documentation in a sense).